variables.tf

```diff
@@ -522,48 +505,38 @@ variable "node_pool_os_version" {
   type        = string
 }
 
-# upgrade of existing node pools
-variable "node_pools_to_drain" {
-  default     = ["none"]
-  description = "List of node pool names to drain during an upgrade. This list is used to determine the worker nodes to drain."
-  type        = list(string)
+variable "worker_mode" {
+  default     = "private"
+  description = "Whether to provision public or private workers."
+  type        = string
 }
 
-variable "nodepool_drain" {
+# upgrade of existing node pools
+variable "upgrade_nodepool" {
   default     = false
   description = "Whether to upgrade the Kubernetes version of the node pools."
   type        = bool
 }
 
+variable "node_pools_to_drain" {
+  default     = ["none"]
+  description = "List of node pool names to drain during an upgrade. This list is used to determine the worker nodes to drain."
+  type        = list(string)
+}
+
 variable "nodepool_upgrade_method" {
   default     = "out_of_place"
   description = "The upgrade method to use when upgrading to a new version. Only out-of-place supported at the moment."
   type        = string
 }
 
-variable "worker_mode" {
-  default     = "private"
-  description = "Whether to provision public or private workers."
-  type        = string
-
-  validation {
-    condition     = contains(["public", "private"], var.worker_mode)
-    error_message = "Accepted values are public or private."
-  }
-}
-
 # oke load balancers
 
 variable "lb_type" {
```

Comment:
the variable naming  is misleading . can it be changed to lbsubnet_type ?

Reply:
good point. I've changed for the preferred one as well.